ui/app/common/displaycontrols/forms/directives/formsTable.js

```diff
@@ -54,15 +79,20 @@ angular.module('bahmni.common.displaycontrol.forms')
 
                 $scope.getDisplayName = function (data) {
                     var concept = data.concept;
-                    var displayName = data.concept.displayString;
-                    if (concept.names && concept.names.length === 1 && concept.names[0].name != "") {
-                        displayName = concept.names[0].name;
-                    } else if (concept.names && concept.names.length === 2) {
-                        var shortName = _.find(concept.names, {conceptNameType: "SHORT"});
-                        displayName = shortName && shortName.name ? shortName.name : displayName;
-                    }
+                    var defaultLocale = $rootScope.currentUser.userProperties.defaultLocale;
+                    var displayName = getLocaleSpecificConceptName(concept, defaultLocale, "FULLY_SPECIFIED");
                     return displayName;
                 };
+                var getLocaleSpecificConceptName = function (concept, locale, conceptNameType) {
```

Comment:
This should be part of the internationalization code, isnt it?

Reply:
Yes. This needs to be applied for forms2 forms too. Hencehere